src/Domain/Neighbors.hpp

```diff
@@ -0,0 +1,95 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Neighbors.
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+#include <unordered_set>
+
+#include "Domain/Direction.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/Orientation.hpp"
+
+/// \ingroup ComputationalDomain
+/// Information about the neighbors of an Element in a particular direction.
+///
+/// \tparam VolumeDim the volume dimension.
+template <size_t VolumeDim>
+class Neighbors {
+  static constexpr int private_asserts() {
+    static_assert(std::is_nothrow_move_constructible<Neighbors>::value and
+                      std::is_nothrow_move_assignable<Neighbors>::value,
+                  "Missing move semantics");
+    return 0;
+  }
+
+ public:
+  /// Construct with the ids and orientation of the neighbors.
+  ///
+  /// \param ids the ids of the neighbors.
+  /// \param orientation the Orientation of the neighbors.
+  Neighbors(std::unordered_set<ElementId<VolumeDim>> ids,
+            Orientation<VolumeDim> orientation);
+
+  /// Default constructor for Charm++ serialization.
+  Neighbors() = default;
+  ~Neighbors() = default;
+  Neighbors(const Neighbors<VolumeDim>& neighbor)  = default;
+  Neighbors(Neighbors<VolumeDim>&&) noexcept = default;  // NOLINT
+  Neighbors<VolumeDim>& operator=(const Neighbors<VolumeDim>& rhs) = delete;
+  Neighbors<VolumeDim>& operator=(Neighbors<VolumeDim>&&) noexcept =  // NOLINT
+      default;                                                        // NOLINT
+
+  const std::unordered_set<ElementId<VolumeDim>>& ids() const { return ids_; }
+
+  const Orientation<VolumeDim>& orientation() const { return orientation_; }
+
+  /// Reset the ids of the neighbors.
+  void set_ids_to(const std::unordered_set<ElementId<VolumeDim>>& new_ids) {
```

Comment:
Nitpicking: is this set or reset? To me, "set" conveys initialization whereas "reset" conveys an update during the simulation.